src/tasks/DSDFarmTask.py

```
@@ -82,35 +107,106 @@ def __init__(self, *args, **kwargs):
+        BaseTask.send_key_up which calls executor.reset_scene() and
+        check_enabled() -- those are unsafe to invoke from the GUI thread
+        (the thread that calls disable()) because they read and mutate
+        executor-internal state (_frame, scene, current_task) that the
+        executor thread may be using at the same moment.
+        """
+        self._stop_requested = True
+        SoundCombatContext().clear_task_if(self)
+        try:
+            for key in ("w", "a", "s", "d", "lshift"):
+                self.executor.interaction.send_key_up(key)
+        except Exception as error:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)